src/js/api.ts

```diff
@@ -14,6 +14,8 @@ import "./error_handling.gen.js";
 import { setStdin, setStdout, setStderr } from "./streams";
 import { makeWarnOnce } from "./util";
 
+import "emscripten";
```

Comment:
where does the "emscripten" import come from? It presumably needs to be in `package.json`. I see:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/emscripten/index.d.ts
but it says they are type definitions for Emscripten 1.39.16 which is way out of date. I guess we could try to bring them up to date if there are in fact any discrepancies. I don't know that the FS API changes much.

Reply:
`@types/emscripten` was already required in [`package.json`](https://github.com/pyodide/pyodide/blob/1691d347d15a2c211cd49aebe6f15d42dfdf2369/src/js/package.json#L20) at the most current major version, so that didn't need to be changed. More current type declarations are definitely needed, admittedly

Reply:
I see, it was introduced in #1692. @bollwyvl does that pr actually use `@types/emscripten`?

Anyways it would be helpful if we could add a bunch of ts tests that use the different functionality of `FS` so that we have an assertion that our type defs reasonably match the runtime behavior. Maybe we could work on that in a followup. what do you think @ryanking13?

Types could definitely use more focused attention but we seem to have trouble ever getting around to working on them with all the other maintenance work around here...

Reply:
That's understandable. Specifically with the `FS` library I found myself constantly going back to the Emscripten API reference so I thought adding type declarations (outdated or not) would at least give me a starting place to work off of.

Reply:
I do think shipping slightly incorrect type declarations is a good way to get bug reports and help gradually fix the problems.

Reply:
Yep: wrong types are often _worse_ than no types, but _maybe_ semi-offically  (or at least _de facto_) wrong types are at least useful. There are a lot of `: any`s down in things at various levels. But having _sort of_ the right shape is a lot more powerful than _just_ any, which is why I opted to bring in those, old as they are. The gross ambient types are not my preference, hence the weird jsdoc construction around [`Module`](https://github.com/pyodide/pyodide/pull/1692/files#diff-29e0ed0a380c63720014afb4aa3b272caddb5e1913c28d0ace55149e2758ba16R4). Something similar can be achieved with _actual_ typescript via, e.g. `typeof Module.FS`.

On the whole, though, this PR seems good, and would be _ready_ to accept improved upstream types, if they existed.

Reply:
I agree that ambient types without a direct import aren't ideal, I’ve been struggling with finding a way to import a specific name from a `dts` file like what's provided in `emscripten` but I think that's a current limitation of TS which is why I landed on `import "emscripten"` however that brings about issues with building. 

In the upcoming days I might try to sort through emscripten's current `FS` library (just at a quick glance I feel it's somewhere [here](https://github.com/emscripten-core/emscripten/blob/main/src/library_fs.js)) and determine if we even have an issue with significantly outdated types against what I can find in the old `@types/emscripten`

Reply:
I didn't realize pyodide already had an interface for `FS`, why isn't that exposed to `PyodideAPI.FS`?